toolchain/driver/clang_runner.cpp

```diff
@@ -42,24 +50,227 @@ auto clang_main(int Argc, char** Argv, const llvm::ToolContext& ToolContext)
 namespace Carbon {
 
 ClangRunner::ClangRunner(const InstallPaths* install_paths,
-                         llvm::StringRef target,
                          llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
                          llvm::raw_ostream* vlog_stream)
     : ToolRunnerBase(install_paths, vlog_stream),
-      target_(target),
       fs_(std::move(fs)),
       diagnostic_ids_(new clang::DiagnosticIDs()) {}
 
+ClangRunner::~ClangRunner() {
+  if (!tmp_dir_.empty()) {
+    std::error_code ec;
+    if (auto number = std::filesystem::remove_all(tmp_dir_, ec); number == 0) {
```

Comment:
Why the temporary here? It seems unused, so maybe just remove it? If you keep it to be used somewhere, consider a name like "count"; "number" is a little too generic.

```suggestion
    if (std::filesystem::remove_all(tmp_dir_, ec) == 0) {
```